app/viewtils/EvidenceFileUploadsHelper.scala

```diff
@@ -41,56 +40,38 @@ class EvidenceFileUploadsHelper @Inject()(govukInsetText: GovukInsetText,
     )
   }
 
-  def duplicateDocumentInset(uploadedFiles: Seq[(UploadJourney, Int)])(
-    implicit messages: Messages, request: UserRequest[_]): Option[Html] = {
+  def getInsetTextForUploadsInRepository(journeyId: String)(implicit messages: Messages): Future[Option[String]] = {
+    uploadJourneyRepository.getUploadsForJourney(Some(journeyId)).map(_.fold[Option[String]](None)(
+      uploads => {
+        val uploadedFiles: Seq[(UploadJourney, Int)] = uploads.filter(file => file.fileStatus == UploadStatusEnum.READY || file.fileStatus == UploadStatusEnum.DUPLICATE).zipWithIndex
+        getInsetTextMessage(uploadedFiles)
+      }
+    ))
+  }
 
-    val mappedDuplicates = uploadedFiles.groupBy(_._1.uploadDetails.map(_.checksum))
+  def getInsetTextForUploads(uploads: Seq[(UploadJourney, Int)])(implicit messages: Messages): Option[Html] = {
+    getInsetTextMessage(uploads).fold[Option[Html]](None)(insetText => Some(govukInsetText(InsetText(content = Text(insetText)))))
+  }
 
+  def getInsetTextMessage(uploadedFiles: Seq[(UploadJourney, Int)])(implicit messages: Messages): Option[String] = {
+    val mappedDuplicates: Map[Option[String], Seq[(UploadJourney, Int)]] = uploadedFiles.groupBy(_._1.uploadDetails.map(_.checksum))
```

Comment:
The boolean (_._2.size > 1), is being evaluated in line 58, 60 and 61 which looks redundant. Can it be stored in a boolean val instead.

Reply:
Good idea